library/src/main/java/com/alamkanak/weekview/WeekView.java

```diff
@@ -152,20 +156,40 @@ public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float d
             if (mIsZooming)
                 return true;
 
-            // Calculate the direction to scroll.
-            if (mCurrentScrollDirection == Direction.NONE) {
-                // allow scrolling only in one direction
-                if (Math.abs(distanceX) > Math.abs(distanceY)){
-                    mCurrentScrollDirection = Direction.HORIZONTAL;
+            switch (mCurrentScrollDirection) {
+                case NONE: {
+                    // allow scrolling only in one direction
+                    if (Math.abs(distanceX) > Math.abs(distanceY)) {
+                        if (distanceX > 0) {
+                            mCurrentScrollDirection = Direction.LEFT;
+                        } else {
+                            mCurrentScrollDirection = Direction.RIGHT;
+                        }
+                    } else {
+                        mCurrentScrollDirection = Direction.VERTICAL;
+                    }
+                    break;
                 }
-                else {
-                    mCurrentScrollDirection = Direction.VERTICAL;
+                case LEFT: {
+                    // change direction if there was enough change
+                    if (Math.abs(distanceX) > Math.abs(distanceY) && (distanceX < -getWidth() / 40)) {
```

Comment:
it is kind of a magic number, i just need a small threshold so there has to be a notable change in scroll direction before changing it. else there are some glitches if you put your hands off the screen and the scroll distance is suddenly 0.1 in other direction.

while writing this maybe the best would be if this threshold is calculated including density.